cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -288,9 +293,10 @@ class device_row_comparator {
  */
 template <typename Comparator, weak_ordering... values>
 struct weak_ordering_comparator_impl {
-  __device__ bool operator()(size_type const lhs, size_type const rhs) const noexcept
+  template <typename... Ts>
+  __device__ bool operator()(Ts&&... args) const noexcept
```

Comment:
These should just be by value. Taking things by reference in heavily layered code like this can lead to local/stack memory usage by ODR-using thread local variables. 
```suggestion
  __device__ bool operator()(Ts... args) const noexcept
```

Reply:
Definitely agree we don't want to take them by reference. I was aiming for "perfect forwarding" here, and thought this would bind in the way we wanted but I was misremembering a rule.